### PR TITLE
adds light to explosions

### PR DESCRIPTION
**Pull request by tyeagg**
## About The Pull Request
add light to explosions, please tm first to see performance impact

## Why It's Good For The Game
they look pretty, easier to determine effect on target with explosions and it doesn't really make much sense how explosions don't create any light

## Changelog
:cl:
add: light to explosions
/:cl:

https://user-images.githubusercontent.com/98577688/166097638-df34bfce-aa1a-45c8-95cd-6b3c4ab8e1a2.mp4

### CODE REVIEW

**Comment by Lumipharon** [comment]
You should probably make them args for explosions so it's easier to tweak. Not all explosions are going to be bright fireballs.

Looks cool though

**Review by BraveMole — CHANGES_REQUESTED** [review]
the light range and/or power should depend on the strength of the explosion i'd say

**Comment by Losenis** [comment]
This is actually nice to see

**Comment by tyeagg** [comment]
https://user-images.githubusercontent.com/98577688/166136059-8885b705-e279-410c-bb73-6483204432d5.mp4

first tat shot didn't explode because aphe doesn't explode on turf

File: code/controllers/subsystem/explosions.dm [hunk 1]
```diff
@@ -86,10 +86,10 @@ SUBSYSTEM_DEF(explosions)
...
-	return SSexplosions.explode(epicenter, devastation_range, heavy_impact_range, light_impact_range, flash_range, flame_range, throw_range, adminlog, silent, smoke, small_animation)
+/proc/explosion(atom/epicenter, devastation_range, heavy_impact_range, light_impact_range, flash_range, flame_range = 0, throw_range, adminlog = TRUE, silent = FALSE, smoke = FALSE, light_range, light_power, light_color, small_animation = FALSE)
+	return SSexplosions.explode(epicenter, devastation_range, heavy_impact_range, light_impact_range, flash_range, flame_range, throw_range, adminlog, silent, smoke, light_range, light_power, light_color, small_animation)
 
-/datum/controller/subsystem/explosions/proc/explode(atom/epicenter, devastation_range, heavy_impact_range, light_impact_range, flash_range, flame_range, throw_range, adminlog, silent, smoke, small_animation)
+/datum/controller/subsystem/explosions/proc/explode(atom/epicenter, devastation_range, heavy_impact_range, light_impact_range, flash_range, flame_range, throw_range, adminlog, silent, smoke, light_range, light_power, light_color, small_animation)
```
Comment (BraveMole) [inline code/controllers/subsystem/explosions.dm:92]: You are not using any of these arguments

File: code/controllers/subsystem/explosions.dm [hunk 2]
```diff
@@ -114,6 +114,11 @@ SUBSYSTEM_DEF(explosions)
 
 	if(max_range >= 6 || heavy_impact_range)
 		new /obj/effect/temp_visual/shockwave(epicenter, max_range)
+
+	var/obj/effect/temp_visual/explosionlight/explosion_obj
+	explosion_obj = new /obj/effect/temp_visual/explosionlight(epicenter, max_range)
+	explosion_obj.light_range = max_range
```
Comment (BraveMole) [inline code/controllers/subsystem/explosions.dm:120]: this should be done in the explosion light constructor

**Review by BraveMole — CHANGES_REQUESTED** [review]
Also this might be too laggy anyway.

**Comment by tyeagg** [comment]
yea im closing this

**Pull request closed without merging**